src/space.cpp

```diff
@@ -1228,9 +1228,10 @@ static argval_t do_space(chunk_t *first, chunk_t *second, int &min_sp)
    if (  !chunk_is_token(first, CT_BRACE_OPEN)
       && !chunk_is_token(first, CT_FPAREN_OPEN)
       && chunk_is_token(second, CT_BRACE_OPEN)
+      && first->parent_type != CT_DECLTYPE
       && second->parent_type == CT_BRACED_INIT_LIST)
    {
-      // 'int{9}' vs 'int {9}'
+      // 'int{9}' vs 'int {9}' but not 'decltype(var){9}' as it is covered by sp_after_decltype
```

Comment:
Honestly... I'm pretty sure I would prefer `sp_type_brace_init_lst` to take precedence. The reason for `sp_after_decltype` is partly for compatibility, and partly to avoid lack of space in e.g. `foo(decltype(x)param)`. However, if `sp_type_brace_init_lst=remove`, I probably *do* want no space in e.g. `decltype(var){9}`, just as I would want no space in e.g. `int{9}`.